stablehlo/tests/interpret_send_recv.mlir

```diff
@@ -0,0 +1,28 @@
+// RUN: stablehlo-translate --interpret -split-input-file %s
+
+module @distribution_ops {
+  func.func public @send(%operand : tensor<2x2xi64>, %token : !stablehlo.token) -> !stablehlo.token {
+    %result = "stablehlo.send"(%operand, %token) {
+      channel_handle = #stablehlo.channel_handle<handle = 1, type = 2>,
+      is_host_transfer = true
+    } : (tensor<2x2xi64>, !stablehlo.token) -> !stablehlo.token
+    return %result : !stablehlo.token
+  }
+  func.func public @recv(%token : !stablehlo.token) -> (tensor<2x2xi64>, !stablehlo.token) {
```

Comment:
Other test ideas:

- single program that does **sequential** `send/receive`, `send/receive` on the **same** channel
- single program that does **parallel** `send/receive`, `send/receive` on the **different** channels
- Send 3 tensors, but receive expects 4 (unless this is handled by generic machinery)

Reply:
Updated the test to include the first two tests. The third test is verified by the `run_parallel` op verifier.